### ISSUE TITLE
go-basic.obo file appears to be missing some parent-child relationships

### ISSUE THREAD

**Issue by keseler** [geneontology/helpdesk#192]
For example, riboflavin biosynthetic process is no longer a child of riboflavin metabolic process:

Feb 18:
[Term]
id: GO:0009231
name: riboflavin biosynthetic process
namespace: biological_process
[...]
is_a: GO:0042364 ! water-soluble vitamin biosynthetic process
is_a: GO:0042727 ! flavin-containing compound biosynthetic process

Feb 11:
[Term]
id: GO:0009231
name: riboflavin biosynthetic process
namespace: biological_process
[...]
is_a: GO:0006771 ! riboflavin metabolic process
is_a: GO:0042364 ! water-soluble vitamin biosynthetic process
is_a: GO:0042727 ! flavin-containing compound biosynthetic process

Strangely, on the current QuickGO website, https://www.ebi.ac.uk/QuickGO/term/GO:0009231, the parent term is still there, and there is no indication in the change log that anything changed.  

Thoughts?  Are there plans to obsolete those parent terms, and this is just preparation for it, or is this a bug?

Thanks,
Ingrid

**Comment by ukemi** [comment]
No. This is a bug, likely due to the new relations concerning primary inputs and outputs. @cmungall it looks like 'has primary input' and 'has primary output' are not subproperties of 'has primary input or output'.

**Comment by ukemi** [comment]
Also pinging @balhoff. Shouldn't 'has primary input or output' also be a child of 'has participant'?

**Comment by balhoff** [comment]
@ukemi I see all those in the RO release. However it looks like the GO `ro_import.owl` has not been updated!

**Comment by balhoff** [comment]
I updated the RO import. Let's take a look after the next successful snapshot (hopefully tonight).

**Comment by cmungall** [comment]
My fault, thanks for the prompt update, Jim

**Comment by ukemi** [comment]
Thanks for taking care of this Jim! I saw this as I was about to head out the door yesterday. I just checked snapshot and the editors' version of the ontology. The inferences are back in the editors' version when I open it in Protege, but I don't see them in the snapshot release. Maybe just a lag? I will be out of the office today, but we should leave this open until we are sure the snapshot files are corrected.

From snapshot  go-basic.obo:

format-version: 1.2
data-version: releases/2019-02-19

[Term]
id: GO:0009231
name: riboflavin biosynthetic process
namespace: biological_process
def: "The chemical reactions and pathways resulting in the formation of riboflavin (vitamin B2), the precursor for the coenzymes flavin mononucleotide (FMN) and flavin adenine dinucleotide (FAD)." [GOC:jl, http://www.indstate.edu/thcme/mwking/vitamins.html]
synonym: "riboflavin anabolism" EXACT []
synonym: "riboflavin biosynthesis" EXACT []
synonym: "riboflavin formation" EXACT []
synonym: "riboflavin synthesis" EXACT []
synonym: "vitamin B2 biosynthesis" EXACT []
synonym: "vitamin B2 biosynthetic process" EXACT []
synonym: "vitamin G biosynthesis" EXACT []
synonym: "vitamin G biosynthetic process" EXACT []
xref: MetaCyc:RIBOSYN2-PWY
xref: Wikipedia:Riboflavin
is_a: GO:0042364 ! water-soluble vitamin biosynthetic process
is_a: GO:0042727 ! flavin-containing compound biosynthetic process

Fron snapshot go-obo (no equivalence axiom):

format-version: 1.2
data-version: releases/2019-02-19

[Term]
id: GO:0009231
name: riboflavin biosynthetic process
namespace: biological_process
def: "The chemical reactions and pathways resulting in the formation of riboflavin (vitamin B2), the precursor for the coenzymes flavin mononucleotide (FMN) and flavin adenine dinucleotide (FAD)." [GOC:jl, http://www.indstate.edu/thcme/mwking/vitamins.html]
synonym: "riboflavin anabolism" EXACT []
synonym: "riboflavin biosynthesis" EXACT []
synonym: "riboflavin formation" EXACT []
synonym: "riboflavin synthesis" EXACT []
synonym: "vitamin B2 biosynthesis" EXACT []
synonym: "vitamin B2 biosynthetic process" EXACT []
synonym: "vitamin G biosynthesis" EXACT []
synonym: "vitamin G biosynthetic process" EXACT []
xref: MetaCyc:RIBOSYN2-PWY
xref: Wikipedia:Riboflavin
is_a: GO:0042364 ! water-soluble vitamin biosynthetic process
is_a: GO:0042727 ! flavin-containing compound biosynthetic process

**Comment by ukemi** [comment]
Still not in snapshot go-edit.obo. But it looks like the file is still from Feruary 19. Pinging @kltm to see if there is a pipeline issue.

**Comment by ukemi** [comment]
It is however still in the official release file.

**Comment by balhoff** [comment]
The snapshot succeeded yesterday, and I do see the new RO axioms in go-plus. If you run the reasoner on go-plus, you can see that 'riboflavin biosynthetic process' is a direct subclass of 'riboflavin metabolic process'. But I don't know why this would not have been asserted during the ontology build reasoning step!

**Comment by balhoff** [comment]
This is fixed in the `2019-02-21` version.

**Comment by keseler** [comment]
I don't think everything has been fixed; here's an example of differences between the ontology files downloaded this morning and Feb. 11:

Feb 22:
[Term]
id: GO:0030632
name: D-alanine biosynthetic process
namespace: biological_process
def: "The chemical reactions and pathways resulting in the formation of D-alanine, the D-enantiomer of the amino acid alanine, i.e (2R)-2-aminopropanoic acid." [CHEBI:15570, GOC:jsg, GOC:mah]
synonym: "D-alanine anabolism" EXACT []
synonym: "D-alanine biosynthesis" EXACT []
synonym: "D-alanine formation" EXACT []
synonym: "D-alanine synthesis" EXACT []
is_a: GO:0046145 ! D-alanine family amino acid biosynthetic process
is_a: GO:0046436 ! D-alanine metabolic process

Feb 11:
[Term]
id: GO:0030632
name: D-alanine biosynthetic process
namespace: biological_process
def: "The chemical reactions and pathways resulting in the formation of D-alanine, the D-enantiomer of the amino acid alanine, i.e (2R)-2-aminopropanoic acid." [CHEBI:15570, GOC:jsg, GOC:mah]
synonym: "D-alanine anabolism" EXACT []
synonym: "D-alanine biosynthesis" EXACT []
synonym: "D-alanine formation" EXACT []
synonym: "D-alanine synthesis" EXACT []
is_a: GO:0046145 ! D-alanine family amino acid biosynthetic process
is_a: GO:0046436 ! D-alanine metabolic process
is_a: GO:0046437 ! D-amino acid biosynthetic process

**Comment by balhoff** [comment]
@keseler you are correct—we identified an additional problem, with some discussion here: https://github.com/geneontology/go-site/issues/999 It relates to our handling of the CHEBI hierarchy.

**Comment by ukemi** [comment]
Yup. We are still tweaking things. Most of the discussion is on this ticket.

https://github.com/geneontology/go-site/issues/999

**Comment by balhoff** [comment]
@keseler please take a look at the latest ontology from 2019-02-27; hopefully all the issues are fixed.

**Comment by keseler** [comment]
GO:0019695 choline metabolic process is no longer a parent of GO:0019285.  Was that intended?

[Term]
id: GO:0019285
name: glycine betaine biosynthetic process from choline
namespace: biological_process
def: "The chemical reactions and pathways resulting in the formation of betaine (N-trimethylglycine) from the oxidation of choline." [GOC:jl]
synonym: "choline oxidation" EXACT [GOC:dph, PMID:23563483]
synonym: "glycine betaine anabolism from choline" EXACT []
synonym: "glycine betaine formation from choline" EXACT []
synonym: "glycine betaine synthesis from choline" EXACT []
synonym: "N-trimethylglycine biosynthesis from choline" EXACT []
synonym: "N-trimethylglycine biosynthetic process from choline" EXACT []
xref: MetaCyc:BETSYN-PWY
xref: MetaCyc:CHOLINE-BETAINE-ANA-PWY
xref: MetaCyc:P542-PWY
is_a: GO:0031456 ! glycine betaine biosynthetic process

**Comment by ukemi** [comment]
Hi @keseler  

We are fixing all of the x biosynthetic process from Y terms. Wanted to do a sanity check to be sure that we wanted them before adding them back. We looked at them yesterday and decided that whenever a chemical is named in the pathway as an input or output, we did want that. We wanted to be sure that they never involved transient intermediates, but we were convinced they weren't. So the bottom line is for terms like this, these will come back. We added a lot of them back yesterday and I will continue over the weekend.
